File: duplicates-android/app/src/main/java/com/github/duplicates/contact/ContactFindTask.java
/*
 * Source file of the Remove Duplicates project.
 * Copyright (c) 2016. All Rights Reserved.
 *
 * The contents of this file are subject to the Mozilla Public License Version
 * 2.0 (the "License"); you may not use this file except in compliance with the
 * License. You may obtain a copy of the License at
 * http://www.mozilla.org/MPL/2.0
 *
 * Contributors can be contacted by electronic mail via the project Web pages:
 *
 * https://github.com/pnemonic78/RemoveDuplicates
 *
 * Contributor(s):
 *   Moshe Waisberg
 *
 */
package com.github.duplicates.contact;

import android.content.Context;

import com.github.duplicates.DuplicateFindTask;
import com.github.duplicates.DuplicateTaskListener;

/**
 * Task to find duplicate contacts.
 *
 * @author moshe.w
 */
public class ContactFindTask extends DuplicateFindTask<ContactItem, ContactViewHolder> {

    public ContactFindTask(Context context, DuplicateTaskListener listener) {
        super(context, listener);
    }

    @Override
    protected ContactProvider createProvider(Context context) {
        return new ContactProvider(context);
    }

    @Override
    public ContactAdapter createAdapter() {
        return new ContactAdapter();
    }

    @Override
    public ContactComparator createComparator() {
        return new ContactComparator();
    }
}

File: duplicates-android/app/src/main/java/com/github/duplicates/contact/ContactFindTask.kt
/*
 * Copyright 2016, Moshe Waisberg
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.github.duplicates.contact

import android.content.Context

import com.github.duplicates.DuplicateFindTask
import com.github.duplicates.DuplicateTask
import com.github.duplicates.DuplicateTaskListener

/**
 * Task to find duplicate contacts.
 *
 * @author moshe.w
 */
class ContactFindTask(context: Context, listener: DuplicateTaskListener<ContactItem, DuplicateTask<ContactItem, Any, Any, List<ContactItem>>>) : DuplicateFindTask<ContactItem, ContactViewHolder>(context, listener) {

    override fun createProvider(context: Context): ContactProvider {
        return ContactProvider(context)
    }

    override fun createAdapter(): ContactAdapter {
        return ContactAdapter()
    }

    override fun createComparator(): ContactComparator {
        return ContactComparator()
    }
}
